heartbeat/monitors/active/http/checkjson.go

```diff
@@ -0,0 +1,173 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/PaesslerAG/gval"
+	"github.com/PaesslerAG/jsonpath"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/libbeat/common/jsontransform"
+	"github.com/elastic/beats/v7/libbeat/conditions"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+type jsonChecker func(interface{}) bool
+type compiledJSONCheck struct {
+	description string
+	check       jsonChecker
+	source      string
+}
+
+func checkJSON(checks []*jsonResponseCheck) (bodyValidator, error) {
+	var conditionChecks []compiledJSONCheck
+	var expressionChecks []compiledJSONCheck
+
+	for _, check := range checks {
+		if check.Expression != "" && check.Condition != nil {
+			return nil, fmt.Errorf("only one of 'expression' and 'condition' can be specified for JSON check '%s'", check.Description)
+		}
+
+		if check.Expression != "" {
+			eval, err := gval.Full(jsonpath.PlaceholderExtension()).NewEvaluable(check.Expression)
+			if err != nil {
+				return nil, fmt.Errorf("could not compile gval expression '%s': %w", check.Expression, err)
+			}
+
+			checkFn := func(d interface{}) bool {
+				matches, err := eval.EvalBool(context.Background(), d)
+				if err != nil {
+					logp.Warn("error matching JSON against boolean expression: '%v', %v: %v", check.Expression, matches, err)
+					return false
+				}
+
+				return matches
+			}
+
+			logp.Warn("Add expression: %s\n", check.Expression)
+			expressionChecks = append(expressionChecks, compiledJSONCheck{
+				description: check.Description,
+				check:       checkFn,
+				source:      check.Expression,
+			})
+		} else if check.Condition != nil {
+			cfgwarn.Deprecate("8.0.0", "JSON conditions are deprecated, use 'expression' instead.")
```

Comment:
I can do a follow-up PR to actually delete this in `master` if we agree we should just move to `expression`

Reply:
I don't have a strong opinion here, Do you see people using conditions for simpler use case?